webapp/login.go

```diff
@@ -0,0 +1,255 @@
+package webapp
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/gob"
+	"net/http"
+	"net/url"
+
+	"github.com/google/go-github/github"
+	"github.com/gorilla/securecookie"
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"golang.org/x/oauth2"
+	ghOAuth "golang.org/x/oauth2/github"
+	"google.golang.org/appengine"
+)
+
+// Copyright 2019 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+func init() {
+	gob.Register(map[string]interface{}{})
```

Comment:
It's unclear to me why we'd need `gob`. Could you add some comments?

Reply:
See https://github.com/gorilla/securecookie (search gob); this is needed in order to for securecookies to encode/decode custom types.

Reply:
`User` is a basic type. If I'm reading the docs correctly, one would only need to register a concrete type for an interface.

Reply:
If I don't register User I got "ERROR: Failed to set session cookie: securecookie: error - caused by: securecookie: error - caused by: gob: type not registered for interface: webapp.User".

My understanding is the concrete type is being encoded and decoded as an interface here https://github.com/gorilla/securecookie/blob/86450627d8e6b88e00ea41f228489b89d08a40de/securecookie.go#L449. Plus User is being wrapped around a struct, it should be a custom type?

Reply:
> My understanding is the concrete type is being encoded and decoded as an interface here https://github.com/gorilla/securecookie/blob/86450627d8e6b88e00ea41f228489b89d08a40de/securecookie.go#L449.

Ahh I see. That's why `securecookie` needs to register all custom types, even though they are concrete (`gob` itself only requires you to register interfaces). Can we have a comment here?